adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -478,11 +478,13 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
     newRdd: RDD[T],
     newPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]] = None): U
 
-  protected def getReferenceRegions(elem: T): Seq[ReferenceRegion]
+  protected def getReferenceRegions(
+    elem: T,
+    stranded: Boolean = false): Seq[ReferenceRegion]
 
-  protected def flattenRddByRegions(): RDD[(ReferenceRegion, T)] = {
+  protected def flattenRddByRegions(stranded: Boolean = false): RDD[(ReferenceRegion, T)] = {
```

Comment:
A small request here. Can we add an abstract field `canBeStranded`, and then log a warning message if `flattenRddByRegions` is called with `(!canBeStranded && stranded)`?